### PR TITLE
Update the README.md

### LINKED ISSUE #6: Add README.md

### PR DESCRIPTION
**Pull request by somniumism**
I added contents related to usage, build and test to README.md.

Fixed #6

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,152 @@
 # crypto [![Build Status](https://travis-ci.com/CodeChain-io/rust-codechain-crypto.svg?branch=master)](https://travis-ci.com/CodeChain-io/rust-codechain-crypto) [![License: AGPL v3](https://img.shields.io/badge/License-AGPL%20v3-blue.svg)](https://www.gnu.org/licenses/agpl-3.0)
 The crypto library used by [CodeChain](https://github.com/CodeChain-io/codechain).
+
+## Usage
```
Comment (sgkim126) [inline README.md:4]: It seems that the commit message is wrong. You added a usage to the README.md file, not the file itself.

Comment (somniumism) [reply README.md:4]: Oh, sorry. I modified it to 'Update the README.md' : )

**Review by ScarletBlue — COMMENTED** [review]
I've made the changes by editing your comment.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,152 @@
...
+
+#### The block cipher modes of operation
+
+CodeChain is using two modes of operation: the [AES-256-CBC](https://tools.ietf.org/html/rfc3602) mode for the network module and [AES-128-CTR](https://tools.ietf.org/html/rfc3686#section-2.1) mode for the keystore. 
+
+Let's use the AES-256-CBC mode to show an example:
```
Comment (kseo) [inline README.md]: AES-256-CBC mode example:

File: README.md [hunk 3]
```diff
@@ -1,2 +1,152 @@
...
+let iv = rng.gen();
+
+let encrypted_data = encrypt(message.as_bytes(), &key, &iv).ok().unwrap();
+let decrypted_data = decrypt(&encrypted_data[..], &key, &iv).ok().unwrap();
+
+assert_ne!(message.as_bytes(), &encrypted_data[..]);
+assert_eq!(message.as_bytes(), &decrypted_data[..]);
+```
+
+Let's use the AES-128-CTR mode to show an example:
```
Comment (kseo) [inline README.md]: AES-128-CTR mode example:

File: README.md [hunk 4]
```diff
@@ -1,2 +1,152 @@
...
+let _ = encrypt_128_ctr(&key, &iv, &plaintext, &mut result);
+assert_ne!(result, plaintext);
+
+let ciphertext = result;
+let _ = decrypt_128_ctr(&key, &iv, &ciphertext, &mut result);
+assert_eq!(result, plaintext);
+```
+
+#### Hash functions
+The rust-codechain-crypto crates provide [cryptographic hash functions](https://en.wikipedia.org/wiki/Cryptographic_hash_function) as follows: [RIPE Message Digest](https://en.wikipedia.org/wiki/RIPEMD), [Secure Hash Algorithms](https://en.wikipedia.org/wiki/Secure_Hash_Algorithms) and [BLAKE2](https://en.wikipedia.org/wiki/BLAKE_(hash_function)#BLAKE2).
```
Comment (kseo) [inline README.md]: The list of hash functions provided is as follows:

File: README.md [hunk 5]
```diff
@@ -1,2 +1,152 @@
...
+
+Let's use each hash function to show an example:
```
Comment (kseo) [inline README.md]: Hash function example:

File: README.md [hunk 6]
```diff
@@ -1,2 +1,152 @@
...
+let hash1 = blake256_with_key([0u8; 0], &[0; 64]);
+let hash2 = blake256_with_key([0u8; 0], &[1; 64]);
+assert_ne!(hash1, hash2);
+```
+
+#### Password hashing: Scrypt
+
+CodeChain is using [scrypt](https://en.wikipedia.org/wiki/Scrypt), which is a password-based [key derivation function](https://en.wikipedia.org/wiki/Key_derivation_function) for password hashing.
+
+Let's use scrypt to show an example:
```
Comment (kseo) [inline README.md]: Scrypt example:

File: README.md [hunk 7]
```diff
@@ -1,2 +1,152 @@
...
+result = derive_key(&password, &salt, n, p, r).unwrap();
+right_bits = [144, 79, 151, 99, 185, 187, 191, 74, 135, 222, 178, 102, 32, 179, 194, 170];
+left_bits = [179, 96, 63, 181, 115, 192, 159, 237, 20, 181, 18, 253, 164, 77, 199, 136];
+assert_eq!(&result.0[..], right_bits);
+assert_eq!(&result.1[..], left_bits);
+```
+
+## Build
+
+Download rust-codechain-crypto code
```
Comment (kseo) [inline README.md]: Download the package

File: README.md [hunk 8]
```diff
@@ -1,2 +1,152 @@
...
+
+```sh
+git clone git@github.com:CodeChain-io/rust-codechain-crypto.git
+cd rust-codechain-crypto
+```
+
+Build in debug mode
```
Comment (kseo) [inline README.md]: Why in debug mode?

File: README.md [hunk 9]
```diff
@@ -1,2 +1,152 @@
...
+
+```sh
+cargo build
+```
+
+## Test
+
+Developers are strongly encouraged to write unit tests for new code and submit new unit tests for old code. Unit tests can be compiled and run with: `cargo test --all`. For more details, please refer to [Unit Tests](https://github.com/CodeChain-io/codechain/wiki/Unit-Tests).
```
Comment (kseo) [inline README.md]: Remove the link here.

**Review by kseo — CHANGES_REQUESTED** [review]

**Comment by somniumism** [comment]
@kseo I modified all the things you said, and I added a content related to the release profile as well as the dev profile. Could you check again?

File: README.md [hunk 10]
```diff
@@ -1,2 +1,159 @@
...
+Build in release mode
+
+```sh
+cargo build --release
+```
+This will produce an executable in the ./target/release directory.
+
+## Test
+
+Developers are strongly encouraged to write unit tests for new code and submit new unit tests for old code. Unit tests can be compiled and run with: `cargo test --all`. For more details, please refer to unit tests.
```
Comment (kseo) [inline README.md]: Remove "For more details, please refer to unit tests." sentence.

Comment (somniumism) [reply README.md]: I modified it.

**Review by kseo — APPROVED** [review]
LGTM

**Pull request merged**